skrub/_join_aggregator.py

```diff
@@ -0,0 +1,405 @@
+                op_dict = {
+                    "mean": pl.col(col).mean().alias(out_name),
+                    "std": pl.col(col).std().alias(out_name),
+                    "sum": pl.col(col).sum().alias(out_name),
+                    "min": pl.col(col).min().alias(out_name),
+                    "max": pl.col(col).max().alias(out_name),
+                    "mode": pl.col(col).mode().alias(out_name),
+                }
+                op = op_dict.get(op_name, None)
+                if op is None:
+                    raise ValueError(
+                        f"Polars operation '{op}' is not supported. "
+                        f"Available: {list(op_dict)}"
+                    )
+                stats.append(op)
+
+                # mode() output needs a flattening post-processing
+                if op_name == "mode":
+                    mode_cols.append(out_name)
+
+            return stats, mode_cols
+
+        num_cols, categ_cols = split_num_categ_cols(table.select(cols_to_agg))
+
+        num_ops, categ_ops = split_num_categ_ops(agg_ops)
+
+        num_ops, num_mode_cols = get_agg_ops(num_cols, num_ops)
+        categ_ops, categ_mode_cols = get_agg_ops(categ_cols, categ_ops)
+
+        all_ops = [*num_ops, *categ_ops]
+        table = table.groupby(cols_to_join).agg(all_ops)
+
+        # flattening post-processing of mode() cols
+        flatten_ops = []
+        for col in [*num_mode_cols, *categ_mode_cols]:
+            flatten_ops.append(pl.col(col).list[0].alias(col))
+        table = table.with_columns(flatten_ops)
+
+        cols_renaming = {
+            col: f"{col}{suffix}" for col in table.columns if col not in cols_to_join
+        }
+        table = table.rename(cols_renaming)
+
+        return table
+
+    def join(self, left, right, left_on, right_on):
+        return left.join(
+            right,
+            how="left",
+            left_on=left_on,
+            right_on=right_on,
+        )
+
+
+class JoinAggregator(BaseEstimator, TransformerMixin):
+    """Perform aggregation on auxiliary dataframes before joining
+    on the base dataframe.
+
+    Apply numerical (mean, std, sum, min, max) and categorical (mode)
+    aggregation operations on the columns to aggregate, selected by dtypes.
+
+    The grouping columns used during the aggregation are the columns used
+    as keys for joining.
+
+    Uses :obj:`~pandas.DataFrame` inputs only.
+
+    Parameters
+    ----------
+    tables : list of tuples
+        List of (dataframe, columns_to_join, columns_to_agg) tuple
+        specifying the auxiliary dataframes and their columns for joining
+        and aggregation operations.
+
+        dataframe : :obj:`~pandas.DataFrame` or str
+            The auxiliary data to aggregate and join.
+            'X' can be provided as a string to perform self-aggregation on
+            the input data.
+
+        columns_to_join : str or array-like
+            Select the columns from the dataframe to use as keys during
```

Comment:
Here you are talking about the column in the base table, no?

Reply:
Ah, ok, I understand: this is the "one to many" key.

We are going to have to work on our API to make it readable / understandable as right now even someone who knows well both pandas and scikit-learn does not easily guess what is going on. I don't have a good suggestion right now, though. Will have to think about it.